torch/nn/modules/module.py

```diff
@@ -654,8 +654,7 @@ def _load_from_state_dict(self, state_dict, prefix, metadata, strict, missing_ke
                     continue
 
                 if isinstance(input_param, Parameter):
-                    # backwards compatibility for serialized parameters
-                    input_param = input_param.data
+                    self._parameters[name] = input_param.to(param.device, param.dtype)
```

Comment:
You are loading a state_dict so parameters changing would be OK I think ?

Reply:
Okay, then it might overwrite exiting requires_grad=True Parameter with a require_grad=False Parameter. And it loses object identity so optimizer defined before this using `module.parameters()` will not work properly anymore.

Reply:
If you're loading a serialised Variable, then you are expecting `requires_grad` to be replaced, which is fine I think. 
Alternatively, we could call "rebuild/copy/clone" function on Parameter, ideally in addition with some sort of `buffer` storage like in `modules`. (Use case scenario: I am writing some pytorch extensions for FP16 mixed mode, and would like to attach metadata to Parameters for serialization)

Reply:
For users that construct state_dict themselves, this will be a BC breaking change. The load_state_dict functionality should is really meant to be used to save/restore raw values. I am fine with adding another flag to the function, that will change the requires_grad flag, but that should be off by default.